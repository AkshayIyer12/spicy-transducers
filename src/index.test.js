import { method, transform, chain, identity } from "./index";

const matchSnapshot = (title, value) =>
  it(`matches snapshot of ${title}`, () => {
    expect(value).toMatchSnapshot();
  });

describe("method", () => {
  matchSnapshot("inc function", method("inc")(2)({ inc: v => v + 1 }));
});

describe("transform", () => {
  matchSnapshot("inc function", transform(v => v + 1)(2));
});

describe("chain", () => {
  matchSnapshot(
    "inc twice",
    chain(method("inc")(1), transform(v => v + 1))({ inc: v => v + 1 })
  );
});

describe("identity", () => {
  matchSnapshot("a number", identity(3));
});
