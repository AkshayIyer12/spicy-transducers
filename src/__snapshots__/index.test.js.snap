// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chain matches snapshot of inc twice 1`] = `3`;

exports[`identity matches snapshot of a number 1`] = `3`;

exports[`method matches snapshot of inc function 1`] = `3`;

exports[`transform matches snapshot of inc function 1`] = `3`;
